src/device/dd/disk.c

```diff
@@ -197,3 +201,187 @@ uint32_t PhysToLBA(const struct dd_disk* disk, uint16_t head, uint16_t track, ui
     }
     return 0xFFFF;
 }
+
+
+
+static unsigned int seek_track_mame(const struct dd_disk* disk,
+    uint32_t cur_tk, uint32_t cur_sec, uint32_t host_secbyte,
+    unsigned char bm_write, unsigned char bm_block,
+    unsigned int* bm_zone, unsigned int* bm_track_offset)
+{
+    static const unsigned int start_offset[] = {
+        0x0000000, 0x05f15e0, 0x0b79d00, 0x10801a0,
+        0x1523720, 0x1963d80, 0x1d414c0, 0x20bbce0,
+        0x23196e0, 0x28a1e00, 0x2df5dc0, 0x3299340,
+        0x36d99a0, 0x3ab70e0, 0x3e31900, 0x4149200
+    };
+
+    static const unsigned int tracks[] = {
+        0x000, 0x09e, 0x13c, 0x1d1, 0x266, 0x2fb, 0x390, 0x425
+    };
+
+    unsigned int error = 0;
+    unsigned int tr_off;
+    unsigned int head_x_8 = ((cur_tk & 0x1000) >> 9);
+    unsigned int track = (cur_tk & 0x0fff);
+
+    /* find track bm_zone */
+    for (*bm_zone = 7; *bm_zone > 0; --*bm_zone) {
+        if (track >= tracks[*bm_zone]) {
+            break;
+        }
+    }
+
+    tr_off = track - tracks[*bm_zone];
+
+    /* set zone and track offset */
+    *bm_zone += head_x_8;
```

Comment:
bm_zone is not updated by other format, is this normal ?

Reply:
When I think about it, bm_zone is not really used outside except for checking if the sector size in one of the registers is correct.
Zone is taken into account for MAME format because it has to calculate per disk zone while SDK / D64 format only depends on LBAToByte.